modules/design_system/components/src/button.rs

```diff
@@ -23,26 +33,315 @@ impl ButtonVariant {
     }
 }
 
-pub fn button(text: impl Into<String>) -> Markup {
-    button_with_variant(text, ButtonVariant::Primary, false)
+// TODO: Create type-safe DataStar action wrapper to replace raw string onclick handlers
+// This would provide compile-time validation for DataStar actions like:
+// - DataStarAction::ConsoleLog(msg)
+// - DataStarAction::Get { url, options }
+// - DataStarAction::Post { url, options }
```

Comment:
as discussed the other day: we an even explore binding these directly to to the server logic, to make it compile-time guaranteed that the routes we encode in datastar actually exist in our backend. But that's for later.